xapian-bindings/java/SmokeTest.java

```diff
@@ -121,8 +121,19 @@ public static void main(String[] args) throws Exception {
 		System.exit(1);
 	    }
 	    Enquire enq = new Enquire(db);
-	    enq.setQuery(new Query(Query.OP_OR, "there", "is"));
+	    enq.setQuery(new Query("dog"));
 	    MSet mset = enq.getMSet(0, 10);
+	    // MSet size is 0
+	    if(mset.size() != 0) {
+	    	for(MSetIterator.MatchDescriptor descriptor : mset) {
+	    		System.err.println("Empty iterator entered invalid section (from MSet java iterator)");
```

Comment:
The overall program wouldn't return non-zero. Should I just add a System.exit(1)? This would make the test cases consistent with the rest of SmokeTest. 
SmokeTest is exiting whenever a test fails (which we don't want). The tests are dependent so it would be difficult to modify it in such a manner that the remaining tests will run even when we encounter failed tests. These issues will be solved when we move to jUnit :).